aten/src/ATen/cpu/vec/vec_half.h

```diff
@@ -0,0 +1,59 @@
+#include <ATen/cpu/vec/intrinsics.h>
+
+#ifdef CPU_CAPABILITY_AVX512
+#if defined(__GNUC__)
+#define __at_align__ __attribute__((aligned(64)))
+#elif defined(_WIN32)
+#define __at_align__ __declspec(align(64))
+#else
+#define __at_align__
+#endif
+#else // CPU_CAPABILITY_AVX512
+#if defined(__GNUC__)
+#define __at_align__ __attribute__((aligned(32)))
+#elif defined(_WIN32)
+#define __at_align__ __declspec(align(32))
+#else
+#define __at_align__
+#endif
+#endif // CPU_CAPABILITY_AVX512
+
+namespace at {
+namespace vec {
+// See Note [CPU_CAPABILITY namespace]
+inline namespace CPU_CAPABILITY {
+
+#if defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)
+inline uint16_t float_to_half(float val) {
```

Comment:
can we use `Half` as the output type, instead of `uint16_t` ?

Reply:
Returning uint16_t is aligned with the behavior of `fp16_ieee_to_fp32_value` called in Half-inl.h. It is easy for `Half::Half(float value)` to use `float_to_half`.